### ISSUE TITLE
[BUG] Significant number of frontend console warnings

### ISSUE THREAD

**Issue by dikunong** [Team-5-Cyan-Camel/Flatmates#82]
As pictured below, the frontend presents a huge number of compilation warnings to the user.

![image](https://user-images.githubusercontent.com/1442037/120913733-6345ce00-c6ed-11eb-8845-60ad296d4685.png)

A lot of these are from `jsx-a11y` - in particular from the improperly structured `<Link>` components in the navbar. By having `<a>` elements within the links with no `href` attribute declared, a bunch of link elements that go to nowhere are generated. While not noticeable to mouse users, these are selectable via the keyboard, which creates a bad user experience for those who are motor-impaired and unable to use a mouse, or are blind and use a screen reader.

Others appear to be from React regarding improper declaration of dependencies in hooks. This makes it harder for developers to understand what values hooks use/modify, and increases the likelihood of bugs occurring. [See this section of the React docs for more information on hook dependencies.](https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies)

**Comment by JChamberlain28** [comment]
Thanks for the concern on this.
While this is defiantly something that needs addressing in future through your feature request #83 , the developers do not see that the warning messages themselves are a bug. As you described, the warning messages are a side effect of the points made in #83  which in itself is not a bug report.

For example Facebook (creators of react) have console warning errors. Its not pretty, but does not cause issues. 
![image](https://user-images.githubusercontent.com/51986824/120921988-c437cb00-c71a-11eb-899c-110e4523dd82.png)

Regardless of its classification it will be addressed

**Comment by OverCry** [comment]
Evidentally, this issue is more related to future development as well as maintainability as this can cause issues later on when this project is extended further.

Regardless, this is a valid point and will be fixed before any other future development
